tests/test_eigen.h

```diff
@@ -63,4 +63,50 @@ PrintTo( Vector2d const& v, ::std::ostream* os )
 
 } // end namespace Eigen
 
+// ----------------------------------------------------------------------------
+template <typename A, typename B>
+bool
+CompareMatrices( A const& m1, B const& m2 )
+{
+  return m1.isApprox( m2 );
+}
+
+// ----------------------------------------------------------------------------
+template < typename T, int M, int N >
+bool
+CompareMatricesWithEpsilonHelper( Eigen::Matrix< T, M, N > const& a,
+                                  Eigen::Matrix< T, M, N > const& b,
+                                  double epsilon )
+{
+  for ( unsigned i = 0; i < M; ++i )
+  {
+    for ( unsigned j = 0; j < N; ++j )
+    {
+      if ( std::abs( a( i, j ) - b( i, j ) ) > epsilon )
+      {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+// ----------------------------------------------------------------------------
+template <typename A, typename B>
+bool
+CompareMatricesWithEpsilon( A const& m1, B const& m2, double epsilon )
+{
+  return CompareMatricesWithEpsilonHelper( m1, m2, epsilon );
+}
+
+#define EXPECT_MATRIX_EQ(a, b) \
+  EXPECT_PRED2( \
```

Comment:
Per F2F, will fix this in a future PR. (Note for posterity: mostly because deferring it saves me another round of rebasing things and fixing up multiple commits that would be affected.)